### DISCUSSION TITLE
Is it possible to clear all the annotations in realtime?

### DISCUSSION THREAD

**Discussion by Songhao-LI** [annotorious/annotorious#477] [Q&A]
for now, I found clearAnnotations() is deleting all the annotations using a loop? So, when the dataset is huge, it may take a long time to remove all the annotations, is there any more efficient ways to solve this problem?

**Comment by rsimon** [comment]
I think you may be referring to this line?
https://github.com/annotorious/annotorious/blob/8eaf2bb53534ff6336aad16ff18d3e3dac260894/packages/annotorious-openseadragon/src/annotation/pixi/PixiLayer.svelte#L153

Followed by
https://github.com/annotorious/annotorious/blob/8eaf2bb53534ff6336aad16ff18d3e3dac260894/packages/annotorious-openseadragon/src/annotation/pixi/stageRenderer.ts#L222

There might be room for optimization & I appreciate PRs. However, keep in mind the loop only updates the Map data structures. The actual redraw (which is the expensive operation) happens only once, after the loop.
